### PR TITLE
feat(postgresql): easier SSL config and options param support

### PR DESCRIPTION
**Pull request by rafiss**
### Pull Request Checklist

_Please make sure to review and check all of these items:_

- [x] Have you added new tests to prevent regressions?
- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
I don't think docs are needed
- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)?

### Description Of Change

This commit uses the pg_connection_string package to parse the
connection string if the dialect is postgresql. This is helpful because
it automatically handles reading SSL certs that are specified in the
connection string.

As part of this, support was added for the `options` URL parameter,
which allows arbitrary session variables to be configured in the
connection string.

fixes https://github.com/sequelize/sequelize/issues/13672

### Todos

None

### CODE REVIEW

File: lib/sequelize.js [hunk 1]
```diff
@@ -232,6 +233,12 @@ class Sequelize {
           }
         }
       }
+
+      // For postgres, we can use this helper to load certs directly from the
+      // connection string.
+      if (options.dialect === 'postgres' || options.dialect === 'postgresql') {
+        Object.assign(options.dialectOptions, pgConnectionString.parse(arguments[0]));
```
Comment (sdepold) [inline lib/sequelize.js:240]: Just a thought really: isn't args[0] === database?

File: test/integration/dialects/postgres/connection-manager.test.js [hunk 2]
```diff
@@ -47,6 +47,13 @@ if (dialect.match(/^postgres/)) {
       const error = await sequelize.query('select pg_sleep(2)').catch(e => e);
       expect(error.message).to.equal('Query read timeout');
     });
+
+    it('should allow overriding session variables through the `options` param', async () => {
```
Comment (sdepold) [inline test/integration/dialects/postgres/connection-manager.test.js:51]: I somehow expected a test that would use a database url as you proposed in the description

Comment (rafiss) [reply test/integration/dialects/postgres/connection-manager.test.js:51]: I would like to add a test like that - but I wasn't sure where before. i just found it in `test/integration/sequelize.test.js` now. I'll add a test for `options` there, but I'm having trouble finding any sort of SSL testing anywhere.

File: lib/sequelize.js (hunk 1 above)
Comment (rafiss) [reply lib/sequelize.js:240]: I don't think so - as per https://github.com/sequelize/sequelize/blob/47c4494968422585bf265063925d1662ffcd4173/lib/sequelize.js#L185

it is treated as a connection string in this conditional branch.

Comment (sdepold) [reply lib/sequelize.js:240]: Yeah I found that line confusing as well :) anyway. I don't mind either way

**Review by sdepold — APPROVED** [review]

**Comment by rafiss** [comment]
Thanks for the review! It looks like the test failures are in MSSQL and MySQL, but they seem unrelated to my change.

**Comment by rafiss** [comment]
Thanks for releasing this! here's a little example that shows how much easier it is to use the connection string without needing to parse it: https://github.com/cockroachdb/examples-orms/pull/144/files#diff-0d2dbddd194015ba8186248f2006b9102c626cb8d27f60fdeafec56dd5d7ccb0

**Comment by sdepold** [comment]
Nice one!

File: lib/sequelize.js (hunk 1 above)
Comment (WikiRik) [reply lib/sequelize.js:240]: Can you check if this still occurs on `main`/the latest v7 alpha? If so, can you create an issue for it?

Comment (ankon) [reply lib/sequelize.js:240]: https://github.com/sequelize/sequelize/issues/15823

Took a while. Issue creation has a lot of requirements :)

**Pull request merged**